cub/test/catch2_test_device_segmented_reduce_env.cu

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "insert_nested_NVTX_range_guard.h"
+
+#include <cub/device/device_segmented_reduce.cuh>
+
+#include <thrust/device_vector.h>
+
+#include <c2h/catch2_test_helper.h>
+
+template <int BlockThreads>
+struct reduce_tuning : cub::detail::reduce::tuning<reduce_tuning<BlockThreads>>
+{
+  template <class /* AccumT */, class /* Offset */, class /* OpT */>
+  struct fn
+  {
+    struct Policy500 : cub::ChainedPolicy<500, Policy500, Policy500>
+    {
+      struct ReducePolicy
+      {
+        static constexpr int VECTOR_LOAD_LENGTH = 1;
+
+        static constexpr cub::BlockReduceAlgorithm BLOCK_ALGORITHM = cub::BLOCK_REDUCE_WARP_REDUCTIONS;
+
+        static constexpr cub::CacheLoadModifier LOAD_MODIFIER = cub::LOAD_DEFAULT;
+
+        static constexpr int ITEMS_PER_THREAD = 1;
+        static constexpr int BLOCK_THREADS    = BlockThreads;
+      };
+
+      using SingleTilePolicy      = ReducePolicy;
+      using SegmentedReducePolicy = ReducePolicy;
+    };
+
+    using MaxPolicy = Policy500;
+  };
+};
+
+struct get_scan_tuning_query_t
+{};
+
+struct scan_tuning
+{
+  [[nodiscard]] _CCCL_NODEBUG_API constexpr auto query(const get_scan_tuning_query_t&) const noexcept
+  {
+    return *this;
+  }
+
+  // Make sure this is not used
+  template <class /* AccumT */, class /* Offset */, class /* OpT */>
+  struct fn
+  {};
+};
+
+using block_sizes = c2h::type_list<cuda::std::integral_constant<int, 32>, cuda::std::integral_constant<int, 64>>;
+
+C2H_TEST("Device segmented sum can be tuned", "[reduce][device]", block_sizes)
+{
+  constexpr int target_block_size = c2h::get<0, TestType>::value;
```

Comment:
With the current test logic, we are not sure if `target_block_size` is being used or not. Can you please in-corporate `target_block_size` logic into num_segments and num_items. for e.g, have one segment (with `target_block_size` as segment size) with all elements initialized to 1, and then compare the result i.e segment sum with block size.

Reply:
@rbourgeois33 Adding the above suggestion does not actually help in testing if `target_block_size` is used in the implementation. For e.g. [device_reduce_env.cu](https://github.com/NVIDIA/cccl/blob/main/cub/test/catch2_test_device_reduce_env.cu#L140) uses a special custom functor to test this logic. We cannot do the same here as this is `Sum()` algorithm and we cannot pass any reduction operators. Apologies for the confusion. We can skip adding this logic.